### PR TITLE
Add vita-luajit package

### PR DESCRIPTION
**Pull request by sharkwouter**
This is used by lpp for the Vita. I'm making this a draft, since it's quite likely it won't build. It requires installing some libraries on the system on which it is build which may not be present. I'll see if I can find a way around that.

### CODE REVIEW

**Comment by sharkwouter** [comment]
Luajit builds some tools for the host so it can use them during compilation. Because of this I had to add an libc6-dev-i386 to the install dependencies step in the workflow. Without those, luajit cannot be build.

**Comment by sharkwouter** [comment]
Something to note, this package will clash with a Lua package if that will ever be added.

**Comment by isage** [comment]
> Something to note, this package will clash with a Lua package if that will ever be added.

That's because luajit headers should be installed in `include/luajit-<version_major.<version_minor>`, as original makefile does. Not into `include/`

File: luajit/VITABUILD [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  "git+https://github.com/hyln9/vita-luajit.git#commit=0f0252ab8615bc4b097da0bfe0bf10d0666a3dcd"
+)
+sha256sums=(
+  'SKIP'
+)
+depends=('vita-libdl')
+
+build() {
+  cd vita-luajit/src
+  make CFLAGS="-DLUAJIT_USE_SYSMALLOC -DLJ_FFI_STUB -DLUAJIT_DISABLE_JIT" HOST_CC="gcc -m32" CROSS=arm-vita-eabi- TARGET=arm BUILDMODE=static TARGET_SYS=PSP2 PREFIX="${VITASDK}/arm-vita-eabi" TARGET_FLAGS="-fno-optimize-sibling-calls"
```
Comment (isage) [inline luajit/VITABUILD:15]: 1) PREFIX on build stage, according to readme, should point to target system prefix (it's one of the paths where lua will search for modules.
2) Don't use `${VITASDK}/arm-vita-eabi`, use `$prefix`

Comment (sharkwouter) [reply luajit/VITABUILD:15]: Oh, you're right the prefix is wrong according to the documentation. Good catch.

**Comment by isage** [comment]
You should probably also install pkgconfig config, so build tools can properly find headers/lib

**Comment by isage** [comment]
Honestly, wouldn't it be easier to patch (or provide your own) top-level makefile, instead of manual copying and long line of build parameters?

**Comment by sharkwouter** [comment]
This is based on the README file for the repo and how it's build in the automated pipeline for [lpp-vita](https://github.com/Rinnegatamante/lpp-vita/blob/b2f5ca985986ee4617cb9e236b5ab6392c087d65/azure-pipelines.yml#L19). Building it in a different way would be possible, but would require extra work and either forking the repo adding a patch file.

**Pull request closed without merging**